src/librustc_mir/interpret/memory.rs

```diff
@@ -576,10 +575,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         }
 
         // # Statics
-        // Can't do this in the match argument, we may get cycle errors since the lock would
-        // be held throughout the match.
-        let alloc = self.tcx.alloc_map.lock().get(id);
-        match alloc {
+        match M::resolve_maybe_global_alloc(self.tcx, &self.extra, id) {
```

Comment:
I don't think this change is necessary. You can just keep fetching the `alloc_map`. The alignment and size won't change between different threads.

Reply:
This change is necessary because at least in some cases `get_size_and_align` is called **before** `get_raw` or `get_raw_mut` and, as a result, the `self.alloc_map` does not have the thread local allocation yet. (`canonical_alloc_id` just reserves the thread local allocation id, but does not do the actual allocation.)

Reply:
Ah sorry, yes, this was under my previous impression of how this worked.